### ISSUE TITLE
Add new duplicate grouping logic

### ISSUE THREAD

**Issue by anaelizabethenriquez** [psu-libraries/researcher-metadata#531]
In addition to the current duplicate grouping logic (updates to which are described in #530), add another way duplicates are grouped, based on DOI.

If a publication has the same DOI as another publication, put them in a duplicate group together. If either or both of the publications are already in duplicate groups, combine the duplicate groups.

Note that we may need to adjust the timing of when the current AI-Pure automerger runs to ensure that this additional grouping doesn't interfere with that automerge process (which will only merge if the duplicate group contains two pubs with one import each). That process probably needs to run after each AI import and before this new grouping process. When we roll out these changes, we should run the revised duplicate grouping process (from #530 ), then run the AI-Pure automerger, then run this new process.

Blocked by #529 and #534
Background discussion is in #523

**Comment by ajkiessl** [comment]
The grouping process is done during the import of publications (for every publication individually) during the Pure, Activity Insight, and OAI (PSU Law) importers.  We can have the AI-Pure automerge do the same thing for just the Pure and AI imports.  The DOI grouping process can then follow in the same way.  We can also then follow up with the DOI automerge and any other automerge.

We should probably still run the automerge cron tasks weekly, just to clean up anything that may not have been merged during import.

Somewhat unrelated question and something to consider: what about the Web of Science publications?  I guess we can hope they get grouped with Pure or AI imports as those come in, but what if there are duplicates within Web of Science that don't match with AI or Pure imports?  There may not be many.  I'm not too familiar with that dataset.  We do have the ability to run a task the goes through every publication in the database and group them.  I think this takes a very long time to complete.  I've never run it to completion before.

**Comment by anaelizabethenriquez** [comment]
Looking at publications from 2015 to present, I see publications with Web of Science imports from before 2019 only. I suspect it's unlikely (but not impossible) that new imports from Pure or AI will have matches for them. Of 2235 publications that have a Web of Science import, 1519 have only a Web of Science import (i.e., haven't been merged with anything else). 

tl;dr -- Once we have finished all of the updates to the grouping and merging processes, I think we should run the task that goes through all publications and tries to group them. And ideally the new duplicate grouping logic in this issue would get included in that task as well.
